django/contrib/admin/tests.py

```diff
@@ -26,18 +57,37 @@ class AdminSeleniumWebDriverTestCase(StaticLiveServerTestCase):
         'django.contrib.sessions',
         'django.contrib.sites',
     ]
-    webdriver_class = 'selenium.webdriver.firefox.webdriver.WebDriver'
+
+    @classmethod
+    def _get_local_webdriver_class(self, spec):
+        """
+        Return the webdriver for a local browser corresponding to the given
+        browser spec.
+        """
+        browsers = {
+            'firefox': 'selenium.webdriver.Firefox',
+            'safari': 'selenium.webdriver.Safari',
+            'opera': 'selenium.webdriver.Opera',
+            'ie': 'selenium.webdriver.Ie',
+            'chrome': 'selenium.webdriver.Chrome',
+            'phantomjs': 'selenium.webdriver.PhantomJS',
+            'edge': 'selenium.webdriver.Edge',
+        }
+        return import_string(browsers[spec])
 
     @classmethod
     def setUpClass(cls):
-        if not os.environ.get('DJANGO_SELENIUM_TESTS', False):
-            raise SkipTest('Selenium tests not requested')
         try:
-            cls.selenium = import_string(cls.webdriver_class)()
+            browser_spec = os.environ.get('DJANGO_SELENIUM_SPECS', '')
+        except:
+            raise SkipTest('Selenium tests not requested.')
+        try:
+            cls.selenium = cls._get_local_webdriver_class(browser_spec)()
         except Exception as e:
-            raise SkipTest('Selenium webdriver "%s" not installed or not '
-                           'operational: %s' % (cls.webdriver_class, str(e)))
-        cls.selenium.implicitly_wait(10)
```

Comment:
Don't know what good this is doing, it made impact neither on the testcases nor the time of execution.